katz/src/main/kotlin/katz/Id.kt

```diff
@@ -0,0 +1,31 @@
+package katz
+
+class Id<out A>(val value: A): HK<Id.F, A> {
+
+    class F
+
+    inline fun <B> map(f: (A) -> B): Id<B> = Id(f(value))
+
+    inline fun <B> flatMap(f: (A) -> Id<B>): Id<B> = f(value)
+
+    companion object : IdInstances
+}
+
+/**
+ * Companion interface for [Id] type
+ * outside due to https://youtrack.jetbrains.com/issue/KT-10532
+ */
+private interface IdInstances {
+    fun <A> HK<Id.F, A>.ev(): Id<A> = this as Id<A>
+
+    fun monad() = object : Monad<Id.F> {
```

Comment:
If that's the case then it needs to be created as top level object. Maybe we can preserve the monad() shortcut by making it return the top level object instead of an inlined one.